graylog2-server/src/main/java/org/graylog2/indexer/retention/strategies/AbstractIndexRetentionStrategy.java

```diff
@@ -92,6 +94,10 @@ private void retainTimeBased(IndexSet indexSet, TimeBasedSizeOptimizingStrategyC
         }
     }
 
+    private boolean indexIsReadOnly(IndicesBlockStatus blockStatus, String index) {
+        return Optional.ofNullable(blockStatus.getIndexBlocks(index)).map(b -> b.contains("index.blocks.write")).orElse(false);
```

Comment:
What about `index.blocks.read_only`?

Reply:
We don't use that. It would also prevent us from writing the closeDate into the metadata field

https://github.com/Graylog2/graylog2-server/blob/master/graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/IndicesAdapterOS2.java#L216-L225